src/tests/EventHistogramWriterTests.cpp

```diff
@@ -0,0 +1,547 @@
+    auto M = createTestMessage(HistogramID, i, DimLengths);
+    auto X = Writer->write(wrapBuilder(M));
+    if (!X.is_OK()) {
+      throw std::runtime_error(X.to_str());
+    }
+    ASSERT_TRUE(X.is_OK());
+  }
+  ++HistogramID;
+  for (size_t i = 0; i < 4; ++i) {
+    auto M = createTestMessage(HistogramID, i, DimLengths);
+    auto X = Writer->write(wrapBuilder(M));
+    if (!X.is_OK()) {
+      throw std::runtime_error(X.to_str());
+    }
+    ASSERT_TRUE(X.is_OK());
+  }
+  ++HistogramID;
+  for (size_t i = 1; i < 4; ++i) {
+    auto M = createTestMessage(HistogramID, i, DimLengths);
+    auto X = Writer->write(wrapBuilder(M));
+    if (!X.is_OK()) {
+      throw std::runtime_error(X.to_str());
+    }
+    ASSERT_TRUE(X.is_OK());
+  }
+  Writer->close();
+  auto Histograms = Group.get_dataset("histograms");
+  hdf5::dataspace::Simple Dataspace(Histograms.dataspace());
+  ASSERT_EQ(Dataspace.current_dimensions().at(0), 3u);
+  ASSERT_EQ(Dataspace.current_dimensions().at(1), 4u);
+  ASSERT_EQ(Dataspace.current_dimensions().at(2), 2u);
+  ASSERT_EQ(Dataspace.current_dimensions().at(3), 2u);
+}
+
+TEST_F(EventHistogramWriter, WriteManyHistograms) {
+  auto File = createFile("Test.EventHistogramWriter.WriteManyHistograms",
+                         FileCreationLocation::Default);
+  auto Group = File.root();
+  auto Writer = Writer::create();
+  Writer->parse_config(createTestWriterTypedJson().dump(), "{}");
+  ASSERT_TRUE(Writer->init_hdf(Group, "{}").is_OK());
+  Writer = Writer::create();
+  Writer->parse_config(createTestWriterTypedJson().dump(), "{}");
+  ASSERT_TRUE(Writer->reopen(Group).is_OK());
+  std::vector<uint32_t> DimLengths{4, 2, 2};
+  for (size_t HistogramID = 0; HistogramID < 18; ++HistogramID) {
+    for (size_t i = 0; i < 4; ++i) {
+      auto M = createTestMessage(HistogramID, i, DimLengths);
+      auto X = Writer->write(wrapBuilder(M));
+      if (!X.is_OK()) {
+        throw std::runtime_error(X.to_str());
+      }
+      ASSERT_TRUE(X.is_OK());
+    }
+  }
+  Writer->close();
+  auto Histograms = Group.get_dataset("histograms");
+  hdf5::dataspace::Simple Dataspace(Histograms.dataspace());
+  ASSERT_EQ(Dataspace.current_dimensions().at(0), 4u);
+  ASSERT_EQ(Dataspace.current_dimensions().at(1), 4u);
+  ASSERT_EQ(Dataspace.current_dimensions().at(2), 2u);
+  ASSERT_EQ(Dataspace.current_dimensions().at(3), 2u);
+
+  std::vector<uint64_t> Buffer(DimLengths.at(0) * DimLengths.at(1) *
+                               DimLengths.at(2));
+  Dataspace.selection(hdf5::dataspace::SelectionOperation::SET,
+                      hdf5::dataspace::Hyperslab({1, 0, 0, 0}, {1, 4, 2, 2}));
+  hdf5::dataspace::Simple SpaceMem({4, 2, 2});
+  Histograms.read(Buffer, Histograms.datatype(), SpaceMem, Dataspace);
+  for (size_t Flat = 0; Flat < Buffer.size(); ++Flat) {
+    ASSERT_EQ(Buffer.at(Flat), getValueAtFlatIndex(17, Flat, DimLengths));
+  }
+}
+
+TEST_F(EventHistogramWriter, WriteMultipleHistogramsWithMinimumInterval) {
```

Comment:
Can we disable it for the moment? 
`TEST_F(DISABLED_EventHistogramWriter, WriteMultipleHistogramsWithMinimumInterval)` should work.